src/controllers/admin-controller/api-key-controller/api-key.controller.ts

```diff
@@ -0,0 +1,159 @@
+import { PermissionService } from '@services/user-management/permission.service';
+import { JwtAuthGuard } from "@auth/jwt-auth.guard";
+import { OrganizationAdmin } from "@auth/roles.decorator";
+import { RolesGuard } from "@auth/roles.guard";
+import { ApiKeyResponseDto } from "@dto/api-key/api-key-response.dto";
+import { CreateApiKeyDto } from "@dto/api-key/create-api-key.dto";
+import { ListAllApiKeysResponseDto } from "@dto/api-key/list-all-api-keys-response.dto";
+import { ListAllApiKeysDto } from "@dto/api-key/list-all-api-keys.dto";
+import { DeleteResponseDto } from "@dto/delete-application-response.dto";
+import { AuthenticatedRequest } from "@dto/internal/authenticated-request";
+import { ApiKey } from "@entities/api-key.entity";
+import { ActionType } from "@entities/audit-log-entry";
+import { ErrorCodes } from "@enum/error-codes.enum";
+import {
+	checkIfUserHasAdminAccessToAllOrganizations,
+    checkIfUserHasAdminAccessToAnyOrganization,
+    checkIfUserHasAdminAccessToOrganization,
+} from "@helpers/security-helper";
+import {
+    Body,
+    Controller,
+    Delete,
+    ForbiddenException,
+    Get,
+    NotFoundException,
+    Param,
+    ParseIntPipe,
+    Post,
+    Query,
+    Req,
+    UseGuards,
+} from "@nestjs/common";
+import {
+    ApiBearerAuth,
+    ApiForbiddenResponse,
+    ApiNotFoundResponse,
+    ApiOperation,
+    ApiTags,
+    ApiUnauthorizedResponse,
+} from "@nestjs/swagger";
+import { ApiKeyService } from "@services/api-key-management/api-key.service";
+import { AuditLog } from "@services/audit-log.service";
+import { OrganizationService } from "@services/user-management/organization.service";
+import { Read, Write } from "@auth/roles.decorator";
+
+@UseGuards(JwtAuthGuard, RolesGuard)
+@ApiBearerAuth()
+@OrganizationAdmin()
```

Comment:
Note that this had **no effect** until the fix in roles.guard.ts was introduced in this branch